### PR TITLE
XW-608 clean up prepare* functions

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: server/app/facets/operations/preparePageData.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import localSettings from '../../../config/localSettings';
+import deepExtend from 'deep-extend';
+
```
Comment (bognix) [inline server/app/facets/operations/preparePageData.js]: I'm not sure if name of this file is the best, although it was the best idea that came to my mind at that time.

File: server/app/facets/operations/preparePageData.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+export function getQualarooScriptUrl(request) {
+	// all the third party scripts we don't want to load on noexternals
+	if (!request.query.noexternals) {
+		// qualaroo
+		if (localSettings.qualaroo.enabled) {
+			return localSettings.qualaroo.scriptUrl;
+		}
+	}
+
+	return false;
```
Comment (bognix) [inline server/app/facets/operations/preparePageData.js]: does someone have a better idea how to write this function so it has consistent return type?

File: server/app/facets/operations/prepareCuratedContentData.js [hunk 3]
```diff
@@ -19,69 +38,25 @@ export default function prepareCuratedContentData(request, curatedContentPageDat
...
-			result.openGraph.image = mainPageDetails.thumbnail;
-		}
-	}
 
+	result.openGraph = getOpenGraphData('website', wikiVariables.siteName, result.canonicalUrl, result.mainPageData);
 	// clone object to avoid overriding real localSettings for future requests
-	result.localSettings = deepExtend({}, localSettings);
+	result.localSettings = getLocalSettings();
 
 	if (typeof request.query.buckySampling !== 'undefined') {
```
Comment (bognix) [inline server/app/facets/operations/prepareCuratedContentData.js]: it turned out that this query param is accessed in two places in front - trackPerf and in common.js#getTracker

File: server/app/facets/operations/preparePageData.js (hunk 2 above)
Comment (hakubo) [inline server/app/facets/operations/preparePageData.js]: well you could return here empty string - but that does not sound great either
you could throw - but we don't do that often in our code

I think false is ok here
at least you can decide easily what to do and fallback onto something when you don't have this url

File: server/app/facets/operations/preparePageData.js (hunk 1 above)
Comment (hakubo) [inline server/app/facets/operations/preparePageData.js]: Page sounds good for now
we might consider renaming it to prepareMWPageData to be more explicit

**Pull request merged**